### ISSUE TITLE
Adding bulk versions for setMeshEdge, setMeshTriangle

### ISSUE THREAD

**Issue by fsimonis** [precice/precice#465]
The preCICE API offers methods to set mesh vertices, edges and triangles.

We currently provide bulk versions only for adding vertices. We should add bulk versions for all primitives.
Bulk functions allow us to sanitize the entire data once at the interface and thus offers great runtime benefits during mesh creation.

### Status

| Primitve | Singular | Bulk |
| --- | --- | --- |
| Vertex | `setMeshVertex`  | `setMeshVertices` |
| Edge | `setMeshEdge` | :x: | 
| Triangle | `setMeshTriangle` | :x: | 
| Quad | `setMeshQuad` |

There are also `XWidthEdges()` versions of these.

### setMeshEdges
`void setMeshEdges(int meshID, int size, int const * vertexIDs, int * edgeIDs)`

`vertexIDs` has to point to a range of `2*size` vertex ids in the format `e0A e0B e1A e1B ...`
`edgeIDs` has to point to a range of `size` edge ids. The function writes the resulting edge ids to it in the format `e0 e1 e2 ...`

### setMeshTriangles
`void setMeshTriangles(int meshID, int size, int const * edgeIDs)`

`edgeIDs` has to point to a range of `3*size` edge ids in the format `t0AB t0BC t0CA t1AB t1BC ...`

### setMeshQuads
`void setMeshQuads(int meshID, int size, int const * edgeIDs)`

`edgeIDs` has to point to a range of `4*size` edge ids in the format `t0AB t0BC t0CD q0DA q1AB q1BC ...`

**Comment by IshaanDesai** [comment]
Me and @BenjaminRueth agreed that there is a use case for the bulk command `setMeshEdges`  in the fenics-adapter to get the `nearest-projection` working more efficiently

**Comment by BenjaminRueth** [comment]
In the fenics adapter we currently use a [work around](https://github.com/precice/fenics-adapter/blob/1d3258f7a20a44183d020e0770bb715452886f1c/fenicsadapter/fenicsadapter.py#L321-L325), since no bulk actions are available:

```
        # Set mesh edges in preCICE to allow nearest-projection mapping
        for i in range(len(edge_vertex_ids1)):
            assert (edge_vertex_ids1[i] != edge_vertex_ids2[i])
            self._interface.set_mesh_edge(self._interface.get_mesh_id(self._config.get_coupling_mesh_name()),
                                          edge_vertex_ids1[i], edge_vertex_ids2[i])
```

**Comment by fsimonis** [comment]
Closed via #1322
